src/connection_string.ts

```diff
@@ -1243,12 +1245,53 @@ export const OPTIONS = {
    * @internal
    * TODO: NODE-5671 - remove internal flag
    */
-  mongodbLogPath: { type: 'any' },
+  mongodbLogPath: {
+    transform({ values: [value] }) {
+      if (
+        !(
+          (typeof value === 'string' && ['stderr', 'stdout'].includes(value)) ||
```

Comment:
When `mongodbLogPath = string that is not 'stdout' | 'stderr'` shouldn't we `default to stderr` instead of throwing `MongoAPIError`.

Do we have MONGODB_LOG_PATH at this point already? We could also check for `mongodbLogPath=undefined` and `default to MONGODB_LOG_PATH value ?? stderr`.

And do I understand correctly that the rest of the cases like `Some point after construction, user-defined stream crashes` and  `mongodbLogPath= 'stdout', but stdout is closed`, etc will be handeled by another sub-task?

Reply:
I apologize for the confusion with the test cases table. I've updated it for clarity and correctness. See [here](https://docs.google.com/document/d/1gDJ9vpVdH7BalRxEvzC_UZz-gIYOLSTU7HTqSoa18FA/edit#heading=h.797de2n1adjo).

The specified type of `mongodbLogPath: 'stderr' | 'stdout' | MongoDBLogWritable`, so we want to throw a type error when the type differs.